src/dynamics/fixed_points/basic.lean

```diff
@@ -71,6 +71,24 @@ calc fb (g x) = g (fa x) : (h.eq x).symm
 protected lemma apply {x : α} (hx : is_fixed_pt f x) : is_fixed_pt f (f x) :=
 by convert hx
 
+@[simp] lemma _root_.set.preimage_iterate_eq {f : α → α} {n : ℕ} :
+  set.preimage (f^[n]) = ((set.preimage f)^[n]) :=
```

Comment:
This lemma looks misplaced. Can you put it in `data.set.basic`?

Reply:
The statement can but the proof cannot because of import order. Of course I could prove it in `data.set.basic` with a longer proof.

Unless we change import ordering, the choice is "good proof, bad location" or "bad proof, good location". I opted for the former but if you prefer I'll switch to the latter.

Reply:
What's missing for the proof? It looks fine to me at first sight.

Reply:
Oops I was talking through my hat: I just needed to add an import. Now done.

Reply:
@semorrison, is this new import spurious to you? (`logic.function.iterate` in `data.set.basic`)

Reply:
I think that's okay. It doesn't bring in anything besides `logic.function.iterate`, and the whole `logic` directory gets compiled pretty early anyway.